src/coreclr/jit/lsraarm64.cpp

```diff
@@ -25,6 +25,415 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    }
+
+    regMaskTP currAvailableRegs      = candidates;
+    regMaskTP overallResult          = RBM_NONE;
+    regMaskTP consecutiveResult      = RBM_NONE;
+    regMaskTP busyRegsInThisLocation = regsBusyUntilKill | regsInUseThisLocation;
+
+// At this point, for 'n' registers requirement, if Rm, Rm+1, Rm+2, ..., Rm+k-1 are
+// available, create the mask only for Rm, Rm+1, ..., Rm+(k-n) to convey that it
+// is safe to assign any of those registers, but not beyond that.
+#define AppendConsecutiveMask(startIndex, endIndex, availableRegistersMask)                                            \
+    regMaskTP selectionStartMask = (1ULL << regAvailableStartIndex) - 1;                                               \
+    regMaskTP selectionEndMask   = (1ULL << (regAvailableEndIndex - registersNeeded + 1)) - 1;                         \
+    consecutiveResult |= availableRegistersMask & (selectionEndMask & ~selectionStartMask);                            \
+    overallResult |= availableRegistersMask;
+
+    DWORD regAvailableStartIndex = 0, regAvailableEndIndex = 0;
+
+    // If we don't find consecutive registers, also track which registers we can pick so
+    // as to reduce the number of registers we will have to spill, to accomodate the
+    // request of the consecutive registers.
+    regMaskTP registersNeededMask = (1ULL << registersNeeded) - 1;
+
+    do
+    {
+        // From LSB, find the first available register (bit `1`)
+        BitScanForward64(&regAvailableStartIndex, static_cast<DWORD64>(currAvailableRegs));
+        regMaskTP startMask = (1ULL << regAvailableStartIndex) - 1;
+
+        // Mask all the bits that are processed from LSB thru regAvailableStart until the last `1`.
+        regMaskTP maskProcessed = ~(currAvailableRegs | startMask);
+
+        // From regAvailableStart, find the first unavailable register (bit `0`).
+        if (maskProcessed == 0)
+        {
+            regAvailableEndIndex = 64;
+            if ((regAvailableEndIndex - regAvailableStartIndex) >= registersNeeded)
+            {
+                AppendConsecutiveMask(regAvailableStartIndex, regAvailableEndIndex, currAvailableRegs);
+            }
+            break;
+        }
+        else
+        {
+            BitScanForward64(&regAvailableEndIndex, static_cast<DWORD64>(maskProcessed));
+        }
+        regMaskTP endMask = (1ULL << regAvailableEndIndex) - 1;
+
+        // Anything between regAvailableStart and regAvailableEnd is the range of consecutive registers available
+        // If they are equal to or greater than our register requirements, then add all of them to the result.
+        if ((regAvailableEndIndex - regAvailableStartIndex) >= registersNeeded)
+        {
+            AppendConsecutiveMask(regAvailableStartIndex, regAvailableEndIndex, (endMask & ~startMask));
+        }
+        currAvailableRegs &= ~endMask;
+    } while (currAvailableRegs != RBM_NONE);
+
+    if ((candidates & 0x8000000100000000) == 0x8000000100000000)
```

Comment:
Instead of these hard-coded values, couldn't you use RBM values, e.g., `RBM_V31 | RBM_V0`?

Reply:
That's a great suggestion. I will fix it.